R/generate_price_data.R

```diff
@@ -37,82 +37,38 @@ get_prices <- function(aScenType) {
 
 #' get_hindcast_prices
 #'
-#' @details Read in FAO prices by country and crop,
-#'          Aggregate to GCAM regions and commodities,
+#' @details Read in FAO prices by GCAM regions and commodity. Prices were processed by Ryna.
 #' @return prices in historical period
 #' @import dplyr
 #' @importFrom readr read_csv
-#' @importFrom tidyr gather
-#' @author KVC October 2017
+#' @author KVC October 2018
 #' @export
 get_hindcast_prices <- function(){
   # Silence package checks
-  prod <- price <- year <- value <- price_fy <- uniqueJoinField <- na.omit <-
-    FAO_country <- GCAM_commodity <- item <- deflator <- NULL
-
-  # Read in mappings
-  agluCtry <- suppressMessages(read_csv(system.file("extdata", "./mappings/AGLU_ctry.csv", package = "gcamland"), skip = 3))
-  iso_GCAM_regID <- suppressMessages(read_csv(system.file("extdata", "./mappings/iso_GCAM_regID.csv", package = "gcamland"), skip = 3))
-  GCAM_region_names <- suppressMessages(read_csv(system.file("extdata", "./mappings/GCAM_region_names.csv", package = "gcamland"), skip = 3))
-  FAO_ag_items_PRODSTAT <- suppressMessages(read_csv(system.file("extdata", "./mappings/FAO_ag_items_PRODSTAT.csv", package = "gcamland"), skip = 3))
-
-  # Read prices & production (we'll weight prices by production when we can aggregate)
-  faoPrices <- suppressMessages(read_csv(system.file("extdata", "./hindcast-data/fao_prices.csv", package = "gcamland")))
-  faoProd <- suppressMessages(read_csv(system.file("extdata", "./hindcast-data/fao_prod.csv", package = "gcamland")))
-
-  # Read in GDP deflator
-  gdpDeflator <- suppressMessages(read_csv(system.file("extdata", "./hindcast-data/gdp_deflator.csv", package = "gcamland"), skip = 1))
+  price <- year <- sector <- GCAM_region_name <- GCAM_commod <- pp_2005usd_tonne <- uniqueJoinField <-  NULL
 
-  # Tidy data
-  faoProd %>%
-    gather(year, prod, -FAO_country, -item) %>%
-    mutate(year = as.integer(year)) ->
-    faoProd
+  # Read prices (these are already aggregated to gcam commodity and region)
+  faoPrices <- suppressMessages(read_csv(system.file("extdata", "./hindcast-data/prod_price_rgn.csv", package = "gcamland"), skip=3))
 
+  # Convert to 1975$/kg, filter for the right region, rename columns
   faoPrices %>%
-    gather(year, price, -FAO_country, -item) %>%
-    mutate(year = as.integer(year)) ->
+    filter(GCAM_region_name == DEFAULT.REGION) %>%
+    rename(sector = GCAM_commod,
+           price = pp_2005usd_tonne) %>%
+    mutate(price = price / 3.05 / 1000) %>%
```

Comment:
I guess 1000 is converting to price per kg.  What is the 3.05 for?

Reply:
3.05 is going from 2005$ to 1975$. I will add a comment.